### PR TITLE
Regular update 2021-05-04

### PR DESCRIPTION
**Pull request by romainx**
Hello,

- Few packages updated
- Fix Hadolint warnings
- Style

Not updated Mambaforge version since it brings a new version of Python `3.9` thanks @mathbunnyru for the heads-up.

- Added a test to check the Python version to avoid this kind of surprise in the future.

Best.

### CODE REVIEW

File: base-notebook/Dockerfile
Comment (mathbunnyru) [inline base-notebook/Dockerfile:54]: Please, fix the style for `&&` several lines above as well.

File: base-notebook/Dockerfile
Comment (mathbunnyru) [inline base-notebook/Dockerfile:56]: I think having separate steps is actually good in this case (there is a clear distinction in the processes), so I would prefer hadolint ignore for this line.

File: base-notebook/Dockerfile
Comment (mathbunnyru) [inline base-notebook/Dockerfile:27]: This change is not actually that simple.
MiniForge has started using python 3.9, so with this change we will switch to python 3.9 as well.
You can see it in the tagging step :)

There is a small release note as well.
https://github.com/conda-forge/miniforge/releases

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

File: base-notebook/Dockerfile
Comment (mathbunnyru) [reply base-notebook/Dockerfile:27]: Maybe we should wait at least till anaconda ships python 3.9 by default?

File: base-notebook/Dockerfile
Comment (mathbunnyru) [reply base-notebook/Dockerfile:56]: After a closer look, it actually makes sense both ways, because several lines above we install locales. So, I'm ok with this change.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]
Let's think about updating python.

File: base-notebook/Dockerfile
Comment (romainx) [reply base-notebook/Dockerfile:54]: Hello,

I do not understand what need to be fixed here regarding the style, sorry.

```dockerfile
fonts-liberation \
run-one && \
apt-get clean && rm -rf /var/lib/apt/lists/* && \
echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
locale-gen
```

File: tensorflow-notebook/Dockerfile
Comment (romainx) [inline tensorflow-notebook/Dockerfile:11]: @mathbunnyru is it the style you want to use? If it's the case I have fixed it.

File: base-notebook/Dockerfile
Comment (romainx) [reply base-notebook/Dockerfile:56]: So if you are Ok :-)

File: base-notebook/Dockerfile
Comment (romainx) [reply base-notebook/Dockerfile:27]: I did not even imagine that the version of Python could change in a patch release. If you want to master the python version and avoid an unwanted change in the future I think we have to write a test. I can do it.

Comment (romainx) [reply base-notebook/Dockerfile:27]: For information, [Miniconda](https://docs.conda.io/en/latest/miniconda.html) (the previous installer is shipped also with Python 3.9 `Miniconda3-py39_4.9.2-Linux-x86_64.sh`.
What was applied here until now was to always use the latest versions of the software and to rely on tags if users want to use a previous version.

**Comment by romainx** [comment]
I'm converted it do draft while waiting for @mathbunnyru approval.

File: base-notebook/Dockerfile
Comment (mathbunnyru) [reply base-notebook/Dockerfile:54]: Several lines above the line start from `&&`like this: `&& apt-get install -yq --no-install-recommends \`

File: tensorflow-notebook/Dockerfile
Comment (mathbunnyru) [reply tensorflow-notebook/Dockerfile:11]: No, I added the comment about the line I didn't like.

File: base-notebook/Dockerfile
Comment (mathbunnyru) [reply base-notebook/Dockerfile:27]: I was also surprised by changing of the Python version in a patch release.

Yes, let's write a test, which detects the current python version and compares it with hardcoded constant - this way we will always know when python is updated.

Ok, let's write a test and update.

File: base-notebook/Dockerfile
Comment (romainx) [reply base-notebook/Dockerfile:54]: Ok got it, I though the comment was related to my change, but Ok I've seen it.

File: tensorflow-notebook/Dockerfile
Comment (romainx) [reply tensorflow-notebook/Dockerfile:11]: Ok because I've rolled back your previous style change and I made the link with your other remark on style with `&&` at the beginning of a line. So I would like to be sure.

File: base-notebook/Dockerfile
Comment (mathbunnyru) [reply base-notebook/Dockerfile:54]: Thanks :)
Sorry for being not 100% clear.

File: tensorflow-notebook/Dockerfile
Comment (mathbunnyru) [reply tensorflow-notebook/Dockerfile:11]: Thanks, right now the style looks good and consistent :)

File: base-notebook/Dockerfile
Comment (romainx) [reply base-notebook/Dockerfile:27]: Sure I will do that 👍

Comment (romainx) [reply base-notebook/Dockerfile]: Test implemented but I rolled back the version of Python to stick to `3.8.x`. We will update it later.

**Comment by romainx** [comment]
@mathbunnyru I've implemented all the changes and also reverted to keep Python `3.8.x`. Please tell me if it's Ok for you now and I will merge it. Thanks.

File: base-notebook/test/test_python.py
Comment (mathbunnyru) [inline base-notebook/test/test_python.py:11]:
```suggestion
    """Check that python version is lower than a max version"""
```

I would also change the word max to the word next.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

File: base-notebook/test/test_python.py
Comment (romainx) [reply base-notebook/test/test_python.py]: Docstring changed, also change the word max to next as you suggested. I hope it will be Ok now.

**Review by mathbunnyru — APPROVED** [review]
👍

**Comment by romainx** [comment]
Ok thanks, I will merge it. 
@mathbunnyru I'm gonna take a week break or a week and a half break, I will be able to spend a limited time here before so if something needs to be reviewed and merged before, tell me.

**Comment by mathbunnyru** [comment]
> Ok thanks, I will merge it.
> @mathbunnyru I'm gonna take a week break or a week and a half break, I will be able to spend a limited time here before so if something needs to be reviewed and merged, tell me.

Could you please take a look in the PR: https://github.com/jupyter/docker-stacks/pull/1277?

**Comment by mathbunnyru** [comment]
Another PR, which is ready (you already took a look and it should be good to go): https://github.com/jupyter/docker-stacks/pull/1288

So, two reviews in total, which can be easily merged.

**Pull request merged**
